### PR TITLE
Update runtime compilation docs for 6.0

### PR DESCRIPTION
**Pull request by pranavkm**

### CODE REVIEW

File: aspnetcore/mvc/views/view-compilation.md [hunk 1]
```diff
@@ -19,6 +19,12 @@ Razor files with a *.cshtml* extension are compiled at both build and publish ti
 ## Razor compilation
 
 Build-time and publish-time compilation of Razor files is enabled by default by the Razor SDK. When enabled, runtime compilation complements build-time compilation, allowing Razor files to be updated if they're edited.
+    
+In addition to build time compilation, .NET 6 support updating Razor views and Razor Pages using <xref:test/hot-reload />
```
Comment (Rick-Anderson) [inline aspnetcore/mvc/views/view-compilation.md:23]: ***TIP*** our `<xref:` never end in `/>`, ie, `<xref:some/uid/path>`

File: aspnetcore/mvc/views/view-compilation.md [hunk 2]
```diff
@@ -19,6 +19,12 @@ Razor files with a *.cshtml* extension are compiled at both build and publish ti
...
+
+::: moniker-end
+
+::: moniker range=">=aspnetcore-3.1 && <=aspnetcore-6.0"
```
Comment (Rick-Anderson) [inline aspnetcore/mvc/views/view-compilation.md:27]:
```suggestion
::: moniker range=">=aspnetcore-3.1 && <=aspnetcore-6.0"
```

**Comment by Rick-Anderson** [comment]
@pranavkm  I forked this to #24180. Is that what you intended? Add hot load comment remove the `## Enable runtime compilation at project creation` section from the .NET 6 version?

**Comment by pranavkm** [comment]
> Add hot load comment remove the ## Enable runtime compilation at project creation section from the .NET 6 version?

Yup!

**Pull request merged**